azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java

```diff
@@ -117,24 +114,22 @@ protected CreateResourceGroupThenCreateNodes(
 
       // This sill create the resource group if it does not exist
       String location = template.getLocation().getId();
-      ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(location);
-      String azureGroupName = resourceGroup.name();
 
-      getOrCreateVirtualNetworkWithSubnet(location, options, azureGroupName);
-      configureSecurityGroupForOptions(group, azureGroupName, template.getLocation(), options);
+      createResourceGroupIfNeeded(group, location, options);
```

Comment:
as you are refactoring it, wouldn't be better to avoid function with side-effects (jclouds doesn't have many of them) and set the `ResourceGroup` on `options` ?